website/project/model.py

```diff
@@ -1989,6 +1989,12 @@ def parent_id(self):
     def project_or_component(self):
         return 'project' if self.category == 'project' else 'component'
 
+    @property
+    def root_id(self):
```

Comment:
I'd prefer not to add this property method, but to use `node.root._id` instead.

`node.root` cannot be `None`, so the conditional here doesn't seem to be necessary.